compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -1423,7 +1423,7 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
         let mut spans_and_needs_box = vec![];
 
         match liberated_sig.output().kind() {
-            ty::Dynamic(predicates, _) => {
+            ty::Dynamic(predicates, _, _) => {
```

Comment:
This should be dyn only

Reply:
Done

Reply:
Did you fix this? I think you still need to replace this and match only `ty::Dyn` here